plugins/inputs/http/http.go

```diff
@@ -235,8 +281,46 @@ func (h *HTTP) gatherURL(
 	return nil
 }
 
+func (h *HTTP) newRequest(url, method, body string) (*http.Request, error) {
+	bodyRC, err := makeRequestBodyReader(h.ContentEncoding, body)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Are you missing the
```suggestion
	defer bodyRC.Close()

```
here?

Reply:
The request isn't sent by function exit, so I thought that closing the body would cause a read error in http client.

Reply:
Well but not closing the body will leak memory I think... :-/